### PR TITLE
Fix pagination errors on categories and topics pages.

### PR DESCRIPTION
**Pull request by baerrus**
- Correct numbering page links
- Correct next/prev links
- Fix topics problem where if PostNumber +1 == TopicPosts, it would not create the second page

### CODE REVIEW

**Comment by baerrus** [comment]
Travis build seems to have failed for unrelated reasons

File: src/controllers/topics.js [hunk 1]
```diff
@@ -53,7 +53,7 @@ topicsController.get = function(req, res, next) {
 
 			var settings = results.settings;
 			var postCount = parseInt(results.topic.postcount, 10);
-			var pageCount = Math.max(1, Math.ceil((postCount - 1) / settings.postsPerPage));
+			var pageCount = Math.max(1, Math.ceil(postCount / settings.postsPerPage));
```
Comment (barisusakli) [inline src/controllers/topics.js]: this causes an empty page at the end of the topic.

**Comment by barisusakli** [comment]
What does this PR fix compared to the current pagination we have? The current version is rendered on the server side as well with proper next prev links AFAIK.

**Comment by baerrus** [comment]
There are several errors in calculating pagination sequence and links. Attaching a picture:
nodebbpaginationerrors

File: src/controllers/topics.js (hunk 1 above)
Comment (baerrus) [inline src/controllers/topics.js]: Are you testing on master? I have tested on master and it was fine.

Comment (barisusakli) [inline src/controllers/topics.js]: Tested again this seems to be fine, wasn't able to reproduce the empty page

**Comment by barisusakli** [comment]
Well thats by design, the current pagination code always shows a link to the first and last page and then shows the pages around the current page. So when you have n pages the pagination shows.

`1,  n-2,  n-1, n,  n+1, n+2, n`

This PR changes it to 
image

Ie always shows 10 links and no links to first and last page.

**Comment by julianlam** [comment]
Yeah, we'd want to show a link to 1st and last page, or at least a new set of arrows for `<<` and `>>`

**Comment by baerrus** [comment]
Are you guys open to getting rid of showing page 10 right after page 3? I mean, no amount of explanation is going to fix User confusion there. With regards to << >> to last and first pages I am with you. It is useful. I can add those too. What do you say?

**Comment by barisusakli** [comment]
Could be changed to what github does maybe?

image

**Comment by baerrus** [comment]
I modeled my pagination on Google - Time tested, simple, they have done tons of testing on that. And users have a strong affinity with that brand

googlepagination

**Comment by nhlpl** [comment]
Sorry, but Google do not show 500 pages of long forum topics. Please keep in mind big board and thanks for an interesting discussion.

**Comment by psychobunny** [comment]
I like the github idea, that's pretty neat

**Comment by baerrus** [comment]
@psychobunny It is neat. That is sort of my problem with it. It is kind of too smart. Users needs something they can act on *without*  stopping to think about.

**Comment by nhlpl** [comment]
Jump to a page is a much better idea as I recently suggested.
https://github.com/NodeBB/NodeBB/issues/3027

Google users usually click only on 3 links on the top of the page. And forum members are heavy users of pagination.

Showing all of the available page numbers can be distracting. Our longest topic has got 759 pages and most of our active users read it from 1 to 759.

**Comment by baerrus** [comment]
@nhlpl  - The option to jump to page is available in the top navigator. I am not changing that or even advocating to change. The infiniscroll is still the default mode. My change applies to infiniscroll = off

File: src/pagination.js [hunk 2]
```diff
@@ -1,67 +1,68 @@
...
 			rel: 'next',
-			href: '?page=' + next
-		});
-	}
+			href: '?page=' + (currentPage+1)
+        });
+    }
 
-	if (currentPage > 1) {
+	if(currentPage > 1){
```
Comment (jillesme) [inline src/pagination.js]: Seems like there's a space missing between if and ( :flushed:

**Pull request closed without merging**